### PR TITLE
Drop pointer for activity id fields

### PR DESCRIPTION
**Pull request by vytautas-karpavicius**
**What changed?**
Drop pointer for activity id fields within internal types.

**Why?**
This is one of PRs planned to drop pointers for always required fields in internal types. It will make them more idiomatic and easier to use. These pointer are legacy of Thrift types that were used previously through our codebase. This will also allow safer migration to grpc, as proto primitive types don't have nils by default.

**How did you test it?**
Passing build and existing tests.

**Potential risks**

### CODE REVIEW

File: host/integration_test.go [hunk 1]
```diff
@@ -2431,7 +2431,7 @@ func (s *integrationSuite) TestDecisionTaskFailed() {
 			return nil, []*types.Decision{{
 				DecisionType: types.DecisionTypeScheduleActivityTask.Ptr(),
 				ScheduleActivityTaskDecisionAttributes: &types.ScheduleActivityTaskDecisionAttributes{
-					ActivityID:                    common.StringPtr(strconv.Itoa(int(1))),
+					ActivityID:                    strconv.Itoa(int(1)),
```
Comment (Groxx) [inline host/integration_test.go:2434]: `"1"` though obviously minor (same with line 2610)

File: host/queryworkflow_test.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ func (s *integrationSuite) TestQueryWorkflow_Sticky() {
 			return nil, []*types.Decision{{
 				DecisionType: types.DecisionTypeScheduleActivityTask.Ptr(),
 				ScheduleActivityTaskDecisionAttributes: &types.ScheduleActivityTaskDecisionAttributes{
-					ActivityID:                    common.StringPtr(strconv.Itoa(int(1))),
+					ActivityID:                    strconv.Itoa(int(1)),
```
Comment (Groxx) [inline host/queryworkflow_test.go:89]: `"1"` a few times

File: host/resetworkflow_test.go [hunk 3]
```diff
@@ -82,7 +82,7 @@ func (s *integrationSuite) TestResetWorkflow() {
 				scheduleActivityCommands = append(scheduleActivityCommands, &types.Decision{
 					DecisionType: types.DecisionTypeScheduleActivityTask.Ptr(),
 					ScheduleActivityTaskDecisionAttributes: &types.ScheduleActivityTaskDecisionAttributes{
-						ActivityID:                    common.StringPtr(strconv.Itoa(i)),
+						ActivityID:                    strconv.Itoa(i),
```
Comment (Groxx) [inline host/resetworkflow_test.go:85]: `"1"`

**Review by Groxx — APPROVED** [review]

File: host/resetworkflow_test.go (hunk 3 above)
Comment (vytautas-karpavicius) [reply host/resetworkflow_test.go:85]: Yep, fixed all of those.

**Pull request merged**